modules/services/window-managers/river.nix

```diff
@@ -0,0 +1,212 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) types;
+  cfg = config.wayland.windowManager.river;
+
+  # Systemd integration
+  variables = builtins.concatStringsSep " " cfg.systemd.variables;
+  extraCommands = builtins.concatStringsSep " "
+    (map (f: "&& ${f}") cfg.systemd.extraCommands);
+  systemdActivation = ''
+    exec "${pkgs.dbus}/bin/dbus-update-activation-environment --systemd ${variables} ${extraCommands}"
+  '';
+
+  toValue = val:
+    if lib.isString val || lib.isDerivation val then
+      toString val
+    else if true == val then
+      "enabled"
+    else if false == val then
+      "disabled"
+    else if lib.isInt val then
+      toString val
+    else if lib.isFloat val then
+      lib.strings.floatToString val
+    else
+      abort "unsupported type ${builtins.typeOf val}";
+
+  # Intermediary function that converts some value (attrs, str, ...) to one or several commands.
+  toArgs = path: value:
+
+    let
+      stringValue = lib.concatStringsSep " " (path ++ [ (toValue value) ]);
+      finalValue = if lib.isAttrs value then
+        toCommand path value
+      else if lib.isList value then
+        lib.lists.flatten (map (x: toArgs path x) value)
+      else if value == null then
+        [ ]
+      else
+        [ stringValue ];
+    in finalValue;
+
+  # toCommand :: [string] -> attrs -> [string]
+  # Recursive function that converts an attrs to a list of commands that can be written to the
+  # config file.
+  toCommand = basePath: attrs:
+    lib.concatLists (lib.mapAttrsToList
+      (key: value: let path = basePath ++ [ key ]; in toArgs path value) attrs);
+
+in {
+  meta.maintainers = [ lib.maintainers.GaetanLepage ];
+
+  options.wayland.windowManager.river = {
+    enable = lib.mkEnableOption "river window manager";
+
+    package = lib.mkPackageOption pkgs "river" {
+      nullable = true;
+      extraDescription = ''
+        Set to <code>null</code> to not add any River package to your path.
+        This should be done if you want to use the NixOS River module to install River.
+      '';
+    };
+
+    xwayland.enable = lib.mkEnableOption "XWayland" // { default = true; };
+
+    systemd = {
+      enable = lib.mkEnableOption null // {
+        default = true;
+        description = ''
+          Whether to enable {file}`river-session.target` on
+          river startup. This links to `graphical-session.target`.
+          Some important environment variables will be imported to systemd
+          and D-Bus user environment before reaching the target, including
+          - `DISPLAY`
+          - `WAYLAND_DISPLAY`
+          - `XDG_CURRENT_DESKTOP`
+          - `NIXOS_OZONE_WL`
+          - `XCURSOR_THEME`
+          - `XCURSOR_SIZE`
+        '';
+      };
+
+      variables = lib.mkOption {
+        type = with types; listOf str;
```

Comment:
Most of those with are not needed, how about using types.listOf instead?

Reply:
You would rather have `type = types.listOf types.str;` ?

Reply:
Yeah I personally prefer avoid using with in some cases.
I have to admit that it is an opinionated feedback and I wouldn't mind if you ignore it.

Reply:
I am fine with this as well.
I have only left the one for the complex `valueType` (`settings` option). Is that fine ?

Reply:
Yeah I think it's cool now, thanks 👍